DotNetHowToIndexers/Program.cs

```diff
@@ -0,0 +1,325 @@
+        /// <remarks>
+        /// See https://docs.microsoft.com/en-us/rest/api/searchservice/create-indexer for more information.
+        /// </remarks>
+        private static IndexingSchedule IndexerSchedule = new IndexingSchedule
+        {
+            Interval = TimeSpan.FromMinutes(30),
+            StartTime = DateTimeOffset.Now
+        };
+
+        private static void Usage()
+        {
+            Console.Error.WriteLine($"Usage: DotNetHowToIndexers ({AzureSqlOption}|{AzureTableStorageOption}|{AzureCosmosDbOption})");
+            Console.Error.WriteLine(AzureSqlOption);
+            Console.Error.WriteLine($"\tSpecify the {AzureSqlConnectionStringProperty} property in {ConfigFilePath}");
+            Console.Error.WriteLine(AzureTableStorageOption);
+            Console.Error.WriteLine($"\tSpecify the {AzureStorageConnectionStringProperty} and {AzureStorageTableNameProperty} properties in {ConfigFilePath}");
+            Console.Error.WriteLine(AzureCosmosDbOption);
+            Console.Error.WriteLine($"\tSpecify the {AzureCosmosDbConnectionStringProperty} and {AzureCosmosDbCollectionNameProperty} properties in {ConfigFilePath}");
+        }
+
+        static void Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                Usage();
+                return;
+            }
+
+            IConfigurationBuilder builder = new ConfigurationBuilder().AddJsonFile(ConfigFilePath);
+            IConfigurationRoot configuration = builder.Build();
+
+            SearchServiceClient serviceClient = CreateSearchServiceClient(configuration);
+
+            string option = args[0];
+            string indexerName;
+            if (option.Equals(AzureSqlOption, StringComparison.OrdinalIgnoreCase))
+            {
+                indexerName = SetupAzureSqlIndexer(serviceClient, configuration);
+            }
+            else if (option.Equals(AzureTableStorageOption, StringComparison.OrdinalIgnoreCase))
+            {
+                indexerName = SetupAzureTableStorageIndexer(serviceClient, configuration);
+            }
+            else if (option.Equals(AzureCosmosDbOption, StringComparison.OrdinalIgnoreCase))
+            {
+                indexerName = SetupAzureCosmosDbIndexer(serviceClient, configuration);
+            }
+            else
+            {
+                Usage();
+                return;
+            }
+
+            Console.WriteLine("Deleting index...");
+            DeleteIndexIfExists(serviceClient);
+
+            Console.WriteLine("Creating index...");
+            CreateIndex(serviceClient);
+
+            Console.WriteLine("Running indexer...");
+            serviceClient.Indexers.Run(indexerName);
+
+            Console.WriteLine("Done...");
+            Console.ReadKey();
+        }
+
+        /// <summary>
+        /// Creates an indexer which pulls data from an Azure SQL database and puts it
+        /// inside the hotels index.
+        /// </summary>
```

Comment:
nit: Some the comments for these helper methods might be a bit of an overkill...